### PR TITLE
Add ability to follow file by path

### PR DESCRIPTION
**Pull request by Kentzo**
- Fix compatibility between python2 and python3
- Make follow to handle truncation (by using heuristic)
- Add mechanism to exit indefinite sleep loop
- Add method to follow file by path that handles re-creation

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,4 +1,3 @@
 include README.rst
-include VERSION.txt
 include LICENSE
-recursive-include src/tailer *
\ No newline at end of file
+recursive-include src/tailer *
```
Comment (six8) [inline MANIFEST.in]: Need to remove `src`

File: fabfile.py [hunk 2]
```diff
@@ -1,28 +0,0 @@
-from fabric.api import local, task, abort, settings
```
Comment (six8) [inline fabfile.py]: This is a convenience for pushing to pypi. If removing it, what's the alternative other than doing a lot of manual commands?

File: tailer/__init__.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+from __future__ import unicode_literals
+
+import io
+import locale
+import logging
+import os
+import re
+import sys
+
+from tailer.version import VERSION
```
Comment (six8) [inline tailer/__init__.py]: `from tailer.version import __version__`

File: tailer/version.py [hunk 4]
```diff
@@ -0,0 +1 @@
+VERSION='0.3.1'
```
Comment (six8) [inline tailer/version.py]: Just call it `__version__`

File: tailer/__init__.py [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+    SEEK_CUR = 1
+    SEEK_END = 2
+else:
+    from io import SEEK_SET, SEEK_CUR, SEEK_END
+
+
+__version__ = VERSION
+
+
+LOG = logging.getLogger('tailer')
```
Comment (six8) [inline tailer/__init__.py]: `log = logging.getLogger(__name__)`

**Comment by six8** [comment]
This looks like a lot of work and a generous PR -- Thanks. It was a little difficult to read the diffs with the file renames and is generally a bigger PR than I'd normally accept, but since this project has been stagnant a while it could use a refresh. After a few nitpicks (shows the quality of the work if I'm only complaining about minor things), this is ready to go. 

If you can squash the commits into one with a good commit message describing all the changes I'd prefer that over many commits. I like it when each PR is just one commit.

**Comment by Kentzo** [comment]
@six8 For my needs I released this PR (with some more changes) as the [tailhead](https://pypi.python.org/pypi/tailhead/1.0.1) package. It's easier to maintain it.

Feel free to modify & apply or reject this PR.

File: fabfile.py (hunk 2 above)
Comment (Kentzo) [inline fabfile.py]: `python setup.py sdist upload` was enough for me.

**Comment by six8** [comment]
@Kentzo I don't want to take credit for your hard work. There's only a few comments that need to be addressed and then the squashing of the commits. I'd rather get this PR merged in and add you as a maintainer of the project than have a fork.

**Comment by Kentzo** [comment]
@six8 I did this work as part as my work. I'll see if I have time to address those issues later.

**Comment by six8** [comment]
@Kentzo

**Pull request closed without merging**